### PR TITLE
🚀 Fix the magical paginate function and sprinkle in a tiny sdk.wait function ⏳✨

### PR DESCRIPTION
**Pull request by mohamedmamdouh22**
I wrapped the recursive call to the function withing try-except block to avoid considering errors in the scraping code as end of the pagination
Also added tiny sdk.wait function to replace page.wait_for_selector, it will just wait for load and the selector

### CODE REVIEW

File: harambe/core.py [hunk 1]
```diff
@@ -189,10 +209,12 @@ async def paginate(
...
-
-                await self._scraper(
-                    self, next_url, self._context
-                )  # TODO: eventually fix this to not be recursive
+                try:
+                    await self._scraper(
+                        self, next_url, self._context
+                    )  # TODO: eventually fix this to not be recursive
+                except (TimeoutError, AttributeError) as e: # if the error is due to a problem with the scraping code
+                    raise Exception(e)
```
[the remark this answers is not in the document]
Comment (awtkns) [reply harambe/core.py:217]: it should catch on line 225 no?

File: harambe/parser/type_currency.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        # Remove currency symbols, commas
+        value = re.sub(r"[^\d.-]", "", value)
+        value = value.strip()
+        # Try to parse the string as a float
+        try:
+            parsed_value = float(value)
+        except ValueError:
+            raise ValueError(f"Unable to parse input as currency: {value}")
+
+        return parsed_value
```
Comment (asim-shrestha) [inline harambe/parser/type_currency.py:29]: good stuff

File: harambe/parser/type_currency.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
+from pydantic import BeforeValidator
+from typing import Any
+import re
+from typing_extensions import Annotated
+
+
+class ParserTypeCurrency:
```
Comment (asim-shrestha) [inline harambe/parser/type_currency.py:7]: Good stuff

File: harambe/parser/type_currency.py [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+        value = str(value).strip()
+
+        # Remove currency symbols, commas
+        value = re.sub(r"[^\d.-]", "", value)
+        value = value.strip()
+        # Try to parse the string as a float
+        try:
+            parsed_value = float(value)
+        except ValueError:
+            raise ValueError(f"Unable to parse input as currency: {value}")
```
Comment (awtkns) [inline harambe/parser/type_currency.py:27]: there is no point re-raising the same error

```suggestion
        parsed_value = float(value)
```

**Review by awtkns — CHANGES_REQUESTED** [review]

File: harambe/core.py [hunk 5]
```diff
@@ -189,10 +212,14 @@ async def paginate(
...
-                    self, next_url, self._context
-                )  # TODO: eventually fix this to not be recursive
+                try:
+                    await self._scraper(
+                        self, next_url, self._context
+                    )  # TODO: eventually fix this to not be recursive
+                except (
+                    Exception
+                ) as e:  # if the error is due to a problem with the scraping code
+                    raise Exception(e)
```
Comment (awtkns) [inline harambe/core.py:222]: lets remove the try catch here its redundant

**Review by awtkns — APPROVED** [review]

**Comment by awtkns** [comment]
please also bump the version in the pyproject.toml so we can release this :)

**Pull request merged**